testament/categories.nim

```diff
@@ -606,62 +614,70 @@ proc runJoinedTest(r: var TResults, cat: Category, testsDir: string) =
-    echo buf
-    quit("megatest execution failed")
-
-  norm buf
-  writeFile("outputGotten.txt", buf)
-  var outputExpected = ""
-  for i, runSpec in specs:
-    outputExpected.add marker & runSpec.file & "\n"
-    outputExpected.add runSpec.output.strip
-    outputExpected.add '\n'
-  norm outputExpected
-
-  if buf != outputExpected:
-    writeFile("outputExpected.txt", outputExpected)
-    discard execShellCmd("diff -uNdr outputExpected.txt outputGotten.txt")
-    echo "output different!"
-    # outputGotten.txt, outputExpected.txt not removed on purpose for debugging.
-    quit 1
-  else:
-    echo "output OK"
-    removeFile("outputGotten.txt")
-    removeFile("megatest.nim")
-  #testSpec r, makeTest("megatest", options, cat)
+  let nSplits = 2
+    # adjust once more tests are added to avoid using too much memory, which
+    # causes some travis instances to fail during link time
+  for j in 0..<nSplits:
+    var megatest: string
+    let specsj = getChunk(specs, nSplits, j)
+    let megatestFile = "megatest$#.nim" % [$j]
+    var megatestExe = absolutePath(megatestFile.changeFileExt ExeExt)
+    echo (megatestFile: megatestFile, j: j, nSplits: nSplits, specsLen: specs.len, blockLen: specsj.len)
+    for i, runSpec in specsj:
+      let file = runSpec.file
+      let file2 = outDir / ("megatest_sub_" & $i & ".nim")
+      # `include` didn't work with `trecmod2.nim`, so using `import`
+      let code = "echo \"" & marker & "\", " & quoted(file) & "\n"
+      createDir(file2.parentDir)
+      writeFile(file2, code)
+      megatest.add "import " & quoted(file2) & "\n"
+      megatest.add "import " & quoted(file) & "\n"
+
+    writeFile(megatestFile, megatest)
+
+    let args = ["c", "--nimCache:" & outDir, "-d:testing", "--listCmd", megatestFile]
+    proc onStdout(line: string) = echo line
+    var (buf, exitCode) = execCmdEx2(command = compilerPrefix, args = args, options = {poStdErrToStdOut, poUsePath}, input = "",
+      onStdout = if verboseMegatest: onStdout else: nil)
+    if exitCode != 0:
+      echo buf
+      quit("megatest compilation failed")
+
+    # Could also use onStdout here.
+    (buf, exitCode) = execCmdEx(megatestExe)
+    if exitCode != 0:
+      echo buf
+      quit("megatest execution failed")
+
+    norm buf
+    writeFile("outputGotten.txt", buf)
+    var outputExpected = ""
+    for i, runSpec in specsj:
+      outputExpected.add marker & runSpec.file & "\n"
+      outputExpected.add runSpec.output.strip
+      outputExpected.add '\n'
+    norm outputExpected
+
+    if buf != outputExpected:
+      writeFile("outputExpected.txt", outputExpected)
+      discard execShellCmd("diff -uNdr outputExpected.txt outputGotten.txt")
+      echo "output different!"
+      # outputGotten.txt, outputExpected.txt not removed on purpose for debugging.
+      quit 1
+    else:
+      echo "output OK"
+      removeFile("outputGotten.txt")
+      removeFile(megatestFile)
+    #testSpec r, makeTest("megatest", options, cat)
```

Comment:
why is here so much diff in categories.nim? AFAIK this file doesn't need to be touched to test compile time FFI.

Reply:
@krux02 
I explained this in top-post: 
> #10611 should be merged first (ignore it in the diff); it's needed because of #10632

Reply:
No it's not needed. As far as I can tell #10611 is unrelated to this PR. Please disconnect this PR from the other one.

Reply:
> No it's not needed.

I just removed it.
As a result, as explained in #10611, I can't test this PR using my own travis instance, so my previous workflow doesn't work anymore:
* A1: test a commit locally; if green:
* A2: push on a private branch; if localy travis is green:
* A3: push on PR branch; this runs on shared travis resources

the point of A2 being to avoid wasting shared travis resources. So I have to keep rebasing/removing that PR now

Reply:
I noted in a previous comment to elaborate what local travis is. This pretty much explais it to me. Still it won't harm if you describe it in the PR anyway. But no you don't need to "keep rebasing/removing", you just push to the branch of the PR and work on shared travis resources. That will simplify your life.